src/lib.rs

```diff
@@ -111,3 +111,82 @@ impl PrivilegeLevel {
         }
     }
 }
+
+/// This structure defines the CPU-internal interrupt vectors.
+///
+/// The values are defined by the following manual sections:
+///   * AMD Volume 2: 8.2
+///   * Intel Volume 3A: 6.3.1
+#[repr(u8)]
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum InterruptVector {
+    /// Divide-by-zero Error
+    DivideByZero = 0x00,
+
+    /// Debug
+    Debug = 0x01,
+
+    /// Non-Maskable Interrupt
+    NonMaskableInterrupt = 0x02,
+
+    /// Breakpoint
+    Breakpoint = 0x03,
+
+    /// Overflow
+    Overflow = 0x04,
+
+    /// Bound Range Exceeded
+    BoundRange = 0x05,
+
+    /// Invalid Opcode
+    InvalidOpcode = 0x06,
+
+    /// Device Not Available
+    DeviceNotAvailable = 0x07,
+
+    /// Double Fault
+    DoubleFault = 0x08,
```

Comment:
I notice for the naming you generally use a shortened version from what we use in `InterruptDescriptorTable`. This is fine, but should we then have this be `Double` for consistency with `Page` below>

Reply:
Fixed.